src/main/kotlin/apply/infra/github/GitHubClient.kt

```diff
@@ -30,22 +34,35 @@ class GitHubClient(
             .accept(MediaType.APPLICATION_JSON)
             .header(HttpHeaders.AUTHORIZATION, bearerToken(gitHubProperties.accessKey))
             .build()
-        val zonedDateTime = endDateTime.atZone(ZoneId.systemDefault())
-        return restTemplate.exchange<List<CommitResponse>>(requestEntity).body
-            ?.filter { it.date <= zonedDateTime }
-            ?.maxByOrNull { it.date }
-            ?.let { Commit(it.hash) }
-            ?: throw IllegalArgumentException("해당 커밋이 존재하지 않습니다. endDateTime: $endDateTime")
+        return runCatching { restTemplate.exchange<List<CommitResponse>>(requestEntity) }
+            .onFailure {
+                when (it) {
+                    is Unauthorized -> throw RuntimeException("유효한 토큰이 아닙니다.")
+                    is Forbidden -> throw RuntimeException("요청 한도에 도달했습니다.")
+                    is NotFound -> throw IllegalArgumentException("PR이 존재하지 않습니다. pullRequestUrl: $pullRequestUrl")
+                    else -> throw RuntimeException("예기치 않은 예외가 발생했습니다.", it)
+                }
+            }
+            .map { it.last(endDateTime) }.getOrThrow()
     }
 
     private fun extract(pullRequestUrl: String): List<String> {
         val result = PULL_REQUEST_URL_PATTERN.find(pullRequestUrl)
-            ?: throw IllegalArgumentException("올바른 형식의 URL이어야 합니다")
+            ?: throw IllegalArgumentException("올바른 형식의 Pull Request URL이어야 합니다")
```

Comment:
c: 아래처럼 바꿔도 좋을 것 같아요!
```suggestion
            ?: throw IllegalArgumentException("올바른 형식의 Pull Request URL이어야 합니다. pullRequestUrl: $pullRequestUrl")
```